41_snapshots-btrfs

```diff
@@ -47,8 +47,7 @@ grub_btrfs_config="${sysconfdir}/default/grub-btrfs/config"
 if ! type btrfs >/dev/null 2>&1; then exit 0; fi # btrfs-progs isn't installed
 [[ -f "${GRUB_BTRFS_MKCONFIG_LIB:-/usr/share/grub/grub-mkconfig_lib}" ]] && . "${GRUB_BTRFS_MKCONFIG_LIB:-/usr/share/grub/grub-mkconfig_lib}" || exit 0 # grub-mkconfig_lib couldn't be found
 # Root filesystem isn't btrfs
-root_fs=$(${grub_probe} --target="fs" / 2>/dev/null)
-[[ "$root_fs" != "btrfs" ]] && exit 0
+[[ "$(${grub_probe} --target="fs" / 2>/dev/null)" != "btrfs" ]] && exit 0
```

Comment:
I don't get it. Isn't the actual problem here that the script exits without error and with code 0 when the root_fs is wrong?
If that `[[ "$(${grub_probe} --target="fs" / 2>/dev/null)" != "btrfs" ]]` result is true,  it would still exit with exit 0 and therefore have a silent failure, wouldn't it?

I am doing something similar [here](https://github.com/Antynea/grub-btrfs/pull/218/files#diff-9b6d1a434bd2a223bf71bbbb1393b35e131311e2a8369779c6be2a116626630fR59), because I also stumbled upon this in the past. Would this also solve your issue?

Reply:
> I don't get it. Isn't the actual problem here that the script exits without error and with code 0 when the root_fs is wrong?

The actual problem here is that the script exits with error and with NOT code 0 when root_fs is wrong, and this causes update-grub to fail too.

> If the $(${grub_probe} --target="fs" / 2>/dev/null) command fails, its result would be different to "btrfs" and therefore it would exit with an error.

Yes, this is correct and expected, but due to `set -e` when grub_probe fails in original code - script is immediately killed by bash, without a chance to handle error. 
This pull request fixes this issue and let script handle this error and exit gracefully with code 0 (which is required by update-grub).

Reply:
Thanks, I understand now. 

> This pull request fixes this issue and allow script to proceed, handle this error and exit gracefully with code 0 (which is required by update-grub):

> > Because I think we should rather use explicit errors instead of built in behavior of bash.
> 
> Yes, I think script should print why it won't work, but also should exit gracefully and let other scripts to work.

So you mean this script should exit with 0 even though it has an error, so update grub still works ever if the script fails?
I know this is not really related to your PR, I am just asking in order to improve my PR.

Reply:
> So you mean this script should exit with 0 even though it has an error, so update grub still works ever if the script fails?

Yes, I think so. 
This script already have this logic, but in original code it is broken and error handling condition is unreachable sometimes. 
Also, it's always better to have bootable on main subvolumes set OS rather than unbootable OS (which can happen if update-grub failed after kernel upgrade).

Reply:
Thank you very much, I updated my PR accordingly. 
Did not come to my mind why it should exit with 0 when there is an error, but makes sense.